bundles/org.openhab.binding.hue/src/main/java/org/openhab/binding/hue/internal/connection/Clip2Bridge.java

```diff
@@ -0,0 +1,1135 @@
+            // on GET / PUT requests the caller handles errors and closes the stream; the session is still OK
+            logger.debug("fatalError() {} {} ignoring", causeId, cause.error);
+        } else if (cause.error == Http2Error.GO_AWAY) {
+            logger.debug("fatalError() {} {} scheduling reconnect", causeId, cause.error);
+
+            // schedule task to open again
+            internalRestartScheduled = true;
+            cancelTask(internalRestartTask, false);
+            internalRestartTask = bridgeHandler.getScheduler().schedule(
+                    () -> internalRestart(onlineState == State.ACTIVE), RESTART_AFTER_SECONDS, TimeUnit.SECONDS);
+
+            // force close immediately to be clean when internalRestart() starts
+            close2();
+        } else {
+            logger.warn("fatalError() {} {} closing", causeId, cause.error, cause);
+            close2();
+        }
+    }
+
+    /**
+     * Method that is called back in case of fatal stream or session events. Schedules fatalError() to be called after a
+     * delay in order to prevent sequencing issues.
+     *
+     * @param listener the entity that caused this method to be called.
+     * @param cause the exception that caused the error.
+     */
+    protected void fatalErrorDelayed(Object listener, Http2Exception cause) {
+        synchronized (fatalErrorTasks) {
+            final int index = fatalErrorTasks.size();
+            fatalErrorTasks.put(index, bridgeHandler.getScheduler().schedule(() -> {
+                fatalError(listener, cause);
+                fatalErrorTasks.remove(index);
+            }, 1, TimeUnit.SECONDS));
+        }
+    }
+
+    /**
+     * HTTP GET a Resources object, for a given resource Reference, from the Hue Bridge. The reference is a class
+     * comprising a resource type and an id. If the id is a specific resource id then only the one specific resource
+     * is returned, whereas if it is null then all resources of the given resource type are returned.
+     *
+     * It wraps the getResourcesImpl() method in a try/catch block, and transposes any HttpUnAuthorizedException into an
+     * ApiException. Such transposition should never be required in reality since by the time this method is called, the
+     * connection will surely already have been authorised.
+     *
+     * @param reference the Reference class to get.
+     * @return a Resource object containing either a list of Resources or a list of Errors.
+     * @throws ApiException if anything fails.
+     * @throws InterruptedException
+     */
+    public Resources getResources(ResourceReference reference) throws ApiException, InterruptedException {
+        sleepDuringRestart();
+        if (onlineState == State.CLOSED) {
+            throw new ApiException("getResources() offline");
+        }
+        try {
+            return getResourcesImpl(reference);
+        } catch (HttpUnauthorizedException e) {
```

Comment:
Perhaps `HttpUnauthorizedException` could derive from `ApiException`, so this rethrowing wouldn't be necessary. Then, callers can choose whether to catch only `ApiException` to handle all, or to also catch `HttpUnauthorizedException` for special handling.

Reply:
Ok. Done.